datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -552,7 +552,9 @@ impl ExternalSorter {
         let fetch = self.fetch;
         let expressions = Arc::clone(&self.expr);
         let stream = futures::stream::once(futures::future::lazy(move |_| {
+            let timer = metrics.elapsed_compute().timer();
```

Comment:
I looked at the rest of this code, and I think there are other non trivial consumers of compute (e.g. the call to `concat_batches` above) 

What I would recommend is moving this timer up into the `SortExec::in_mem_sort` and add one in  `SortExec::in_mem_sort_stream`

Reply:
I did double check and the `streaming_merge` code does seem to track elapsed compute reasonably: https://github.com/apache/datafusion/blob/37428bb2034abc1aba5d4cd52cf8f2885623bada/datafusion/physical-plan/src/sorts/merge.rs#L170-L173

Reply:
Where I have put the timer is part of the stream, so would not execute until the stream is polled, that means it wouldn't be tracked if the timer was moved to `in_mem_sort_stream`. I agree there should probably be another one there too though. `in_mem_sort` looks to be reasonably trivial to me, but I could add timers in the non-async blocks too if you think it's worthwhile.

Reply:
Ah, I think I missed the fact the timer is in the future 👍